antarest/eventbus/service.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import threading
+import time
+from typing import List, Callable, Optional, Dict
+from uuid import uuid4
+
+from antarest.common.interfaces.eventbus import Event, IEventBus
+from antarest.eventbus.business.interfaces import IEventBusBackend
+
+logger = logging.getLogger(__name__)
+
+
+class EventBusService(IEventBus):
+    def __init__(
+        self, backend: IEventBusBackend, autostart: bool = True
+    ) -> None:
+        self.backend = backend
+        self.listeners: Dict[str, Callable[[Event], None]] = {}
+        self.lock = threading.Lock()
+        if autostart:
+            self.start()
+
+    def push(self, event: Event) -> None:
+        # TODO add arg permissions with group/role, user, public
+        self.backend.push_event(event)
```

Comment:
Why events are not send directly to websocket ? What I understand, events are stored inside stack. And each 0.2s stack is clear and event are send to websocket.